### PR TITLE
Test Certificate by Name

### LINKED ISSUE #3293: [Feature Request] Update Lab Access Cert find method
**Is your feature request related to a problem?**
Currently the [AutomationTestAgentCert](https://ms.portal.azure.com/#@microsoft.onmicrosoft.com/asset/Microsoft_Azure_KeyVault/Certificate/https://buildautomation.vault.azure.net/certificates/AzureADIdentityDivisionTestAgentCert) auto renews every 6 months. When this happens it breaks the build because we search based on thumbprint. 

Existing way to find lab [cert](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/a268f994d1458ea76a6fd22b276c670b5ab85856/tests/Microsoft.Identity.Test.LabInfrastructure/CertificateHelper.cs#L47)

**Describe the solution you'd like**
* Following items are tracked for this
    * [ ] Recreate the "AzureADIdentityDivisionTestAgentCert" with a valid Subject Name and Issuer
    * [ ] Update to search based on FindCertificateBySubjectName ([refer this for locations](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/3291))
    * [ ] Cert also needs to be updated on our automation test App (16dab2ba-145d-4b1b-8569-bf4b9aed4dc8)
    * [ ] The PPE tenant app (OBOClientPpeClientID = "9793041b-9078-4942-b1d2-babdc472cc0c") is in a private tenant, move this to the lab owned tenant. OBO App : c84e9c32-0bc9-4a73-af05-9efe9982a322

**Additional context**
This will reduce downtime for our CI pipeline

### PR DESCRIPTION
**Pull request by SameerK-MSFT**
Fixes #3293

**Changes proposed in this request**
Retrieve certificate by name

**Testing**
Lab tests

**Performance impact**
None

**Documentation**
- [x] All relevant documentation is updated.

### CODE REVIEW

File: tests/Microsoft.Identity.Test.LabInfrastructure/LabAuthenticationHelper.cs [hunk 1]
```diff
@@ -72,9 +73,9 @@ public static async Task<AccessToken> GetLabAccessTokenAsync(string authority, s
             {
                 case LabAccessAuthenticationType.ClientCertificate:
                     var clientIdForCertAuth = String.IsNullOrEmpty(clientId) ? LabAccessConfidentialClientId : clientId;
-                    var certThumbprintForLab = String.IsNullOrEmpty(clientId) ? LabAccessThumbPrint : certThumbprint;
+                    // var certThumbprintForLab = String.IsNullOrEmpty(clientId) ? LabAccessThumbPrint : certThumbprint;
```
Comment (bgavrilMS) [inline tests/Microsoft.Identity.Test.LabInfrastructure/LabAuthenticationHelper.cs:76]: not sure you can remove this. There are actually 2 certs that are being used by our automation.

Comment (SameerK-MSFT) [reply tests/Microsoft.Identity.Test.LabInfrastructure/LabAuthenticationHelper.cs]: There are two callers to the method. Both are passing String.Empty to certThumbprint. No point in having it.

File: tests/Microsoft.Identity.Test.LabInfrastructure/CertificateHelper.cs [hunk 2]
```diff
@@ -9,16 +9,16 @@ namespace Microsoft.Identity.Test.LabInfrastructure
     public static class CertificateHelper
     {
         /// <summary>
-        /// Try and locate a certificate matching the given <paramref name="thumbprint"/> by searching in
-        /// the <see cref="StoreName.My"/> store name for all available <see cref="StoreLocation"/>s.
+        /// Try and locate a certificate matching the given <paramref subjectName="subjectName"/> by searching in
```
Comment (neha-bhargava) [inline tests/Microsoft.Identity.Test.LabInfrastructure/CertificateHelper.cs:12]:
```suggestion
        /// Try and locate a certificate matching the given <paramref name="subjectName"/> by searching in
```

File: tests/Microsoft.Identity.Test.LabInfrastructure/CertificateHelper.cs [hunk 3]
```diff
@@ -9,16 +9,16 @@ namespace Microsoft.Identity.Test.LabInfrastructure
...
+        /// the <see cref="StoreName.My"/> store subjectName for all available <see cref="StoreLocation"/>s.
         /// </summary>
-        /// <param name="thumbprint">Thumbprint of certificate to locate</param>
-        /// <returns><see cref="X509Certificate2"/> with <paramref name="thumbprint"/>, or null if no matching certificate was found</returns>
-        public static X509Certificate2 FindCertificateByThumbprint(string thumbprint)
+        /// <param subjectName="subjectName">Thumbprint of certificate to locate</param>
```
Comment (neha-bhargava) [inline tests/Microsoft.Identity.Test.LabInfrastructure/CertificateHelper.cs:15]:
```suggestion
        /// <param name="subjectName">Thumbprint of certificate to locate</param>
```

File: tests/Microsoft.Identity.Test.LabInfrastructure/CertificateHelper.cs [hunk 4]
```diff
@@ -27,32 +27,29 @@ public static X509Certificate2 FindCertificateByThumbprint(string thumbprint)
 
             return null;
         }
-
         /// <summary>
-        /// Try and locate a certificate matching the given <paramref name="thumbprint"/> by searching in
+        /// Try and locate a certificate matching the given <paramref subjectName="certName"/> by searching in
```
Comment (neha-bhargava) [inline tests/Microsoft.Identity.Test.LabInfrastructure/CertificateHelper.cs:31]:
```suggestion
        /// Try and locate a certificate matching the given <paramref name="certName"/> by searching in
```

Comment (neha-bhargava) [reply tests/Microsoft.Identity.Test.LabInfrastructure/CertificateHelper.cs:31]: Here and all other param comments

File: tests/Microsoft.Identity.Test.LabInfrastructure/LabAuthenticationHelper.cs [hunk 5]
```diff
@@ -72,9 +70,8 @@ public static async Task<AccessToken> GetLabAccessTokenAsync(string authority, s
             {
                 case LabAccessAuthenticationType.ClientCertificate:
                     var clientIdForCertAuth = String.IsNullOrEmpty(clientId) ? LabAccessConfidentialClientId : clientId;
-                    var certThumbprintForLab = String.IsNullOrEmpty(clientId) ? LabAccessThumbPrint : certThumbprint;
 
-                    cert = CertificateHelper.FindCertificateByThumbprint(certThumbprintForLab);
+                    cert = CertificateHelper.FindCertificateByName(LabAccessCertName);
```
Comment (pmaytak) [inline tests/Microsoft.Identity.Test.LabInfrastructure/LabAuthenticationHelper.cs:74]: Can `TestConstants.AutomationTestCertName` be used here, just so the cert is specified once in our code?

**Review by pmaytak — APPROVED** [review]

File: tests/Microsoft.Identity.Test.Common/TestConstants.cs [hunk 6]
```diff
@@ -31,7 +31,7 @@ public static HashSet<string> s_scope
         public static readonly string[] s_graphScopes = new[] { "user.read" };
         public const uint JwtToAadLifetimeInSeconds = 60 * 10; // Ten minutes
         public const string ClientCredentialAudience = "https://login.microsoftonline.com/72f988bf-86f1-41af-91ab-2d7cd011db47/v2.0";
-        public const string AutomationTestThumbprint = "444B697D869032F29F9A162D711AF3E2791AD748";
+        public const string AutomationTestCertName = "AzureADIdentityDivisionTestAgentCert";
```
Comment (gladjohn) [inline tests/Microsoft.Identity.Test.Common/TestConstants.cs:34]: if the lab access cert `LabAccessCertName` is used for all tests, no need to store it on `AutomationTestCertName` as well. Let's just use a common name to refer the same cert

File: tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/PoPTests.cs [hunk 7]
```diff
@@ -492,7 +492,8 @@ public async Task WamUsernamePasswordRequestWithPOPAsync()
 
         private static X509Certificate2 GetCertificate()
         {
-            X509Certificate2 cert = CertificateHelper.FindCertificateByThumbprint(TestConstants.AutomationTestThumbprint);
+            X509Certificate2 cert = CertificateHelper.FindCertificateByName(TestConstants.AutomationTestCertName);
```
Comment (gladjohn) [inline tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/PoPTests.cs:495]: nice!!! but, let's not remove `FindCertificateByThumbprint` method from our code completely.

**Review by gladjohn — APPROVED** [review]

**Comment by bgavrilMS** [comment]
@SameerK-MSFT - looks like you have enough approvals, consider merging this after solving comments.

File: tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/PoPTests.cs (hunk 7 above)
Comment (trwalke) [reply tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/PoPTests.cs:495]: Why not remove it, We are only looking for the subject name now.

**Review by trwalke — APPROVED** [review]
LGTM

File: tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/PoPTests.cs (hunk 7 above)
Comment (bgavrilMS) [reply tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/PoPTests.cs:495]: Simple way to tell if you should remove a method - are there 0 references to it? If yes, then remove it :)

File: tests/Microsoft.Identity.Test.LabInfrastructure/LabAuthenticationHelper.cs (hunk 5 above)
Comment (SameerK-MSFT) [reply tests/Microsoft.Identity.Test.LabInfrastructure/LabAuthenticationHelper.cs:74]: Good catch.Done.

File: tests/Microsoft.Identity.Test.Common/TestConstants.cs (hunk 6 above)
Comment (SameerK-MSFT) [reply tests/Microsoft.Identity.Test.Common/TestConstants.cs:34]: AutomationTestCertName is used at more than one places. Removed LabAccessCertName

**Pull request merged**